### PR TITLE
improve the autogenerated doc

### PR DESCRIPTION
**Pull request by cornelurian**
- keep only the component `name` from the header
- remove `ComponentLinks` and `carbon-ad` section

### CODE REVIEW

File: src/enhance.ts [hunk 1]
```diff
@@ -79,11 +87,14 @@ export const enhanceDoc = (chakraDoc: string = ''): Promise<string> => {
...
-  const enhanced = chakraDoc.replaceAll(codeRegex, (_, codeBlock) => {
+  const enhanced = chakraDoc.replace(
+    headerRegex, (headerBlock) => headerBlock && `# ${headerBlock.match(componentNameRegex)}`
+  ).replaceAll(new RegExp(ignoredComponentsRegex, 'gmi'), ''
+  ).replaceAll(codeRegex, (_, codeBlock) => {
     const components: string[] = uniq(
       [...codeBlock.matchAll(componentsRegex)].map(
         ([_, component]) => component
-      )
+      ).filter(component => !ignoredComponentList.includes(component))
```
Comment (mveres) [inline src/enhance.ts]: this should not be needed since you already removed the ignored components and this only looks in the code blocks

**Review by mveres — APPROVED** [review]
except the filter, lgtm

File: src/enhance.ts [hunk 2]
```diff
@@ -16,6 +16,9 @@ code={\`
 $code\`}
 />`;
 
+const headerRegex = /(---)(.*---)/gms;
```
Comment (gluck) [inline src/enhance.ts]: Regex shouldn't be global (`g`) as only the first match should be replaced

File: src/enhance.ts [hunk 3]
```diff
@@ -16,6 +16,9 @@ code={\`
...
+const componentNameRegex = /(?<=title:).*/g;
+const selfAndNormalClosingTag = `<$name[^]*?(\\/>|<\\/$name>)`;
```
Comment (gluck) [inline src/enhance.ts:21]: Regex will fail if there's a self closing tag inside an ignored component, can that happen? If yes prefer `<$name[^/]*(/>|.*</$name>)`

**Review by gluck — APPROVED** [review]

File: src/enhance.ts (hunk 2 above)
Comment (cornelurian) [reply src/enhance.ts]: meanwhile, after I found some parsing issues, I changed it to `/---[^]*?(---)/`

File: src/enhance.ts (hunk 3 above)
Comment (cornelurian) [reply src/enhance.ts:21]: I'm not sure about this case - self closing tag inside an ignored component (not found this scenario till now). At this moment, the `carbon-ad` and `ComponentLinks` are the reason for this.

**Pull request merged**